docs/providers.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Providers
+toc: true
+weight: 400
+indent: true
+---
+
+# Providers
+
+Providers are external infrastructure providers such as Amazon Web Services,
+Microsoft Azure, Google Cloud and Alibaba Cloud.
+
+In order to provision a resource in the provider, a Custom Resource Definition(CRD)
+needs to be registered in your Kubernetes cluster and its controller should
+be watching instances of that Custom Resource Definition. Provider packages
```

Comment:
Ranty nitpick: I'd really love to move away from the "instance of a CRD" terminology. To me an "instance of a CRD" sounds like we're discussing a resource of `kind: CustomResourceDefinition`, not an instance of the custom resource defined by the CRD.